controllers/amazoncloudwatchagent_controller.go

```diff
@@ -40,6 +43,80 @@ type Params struct {
 	Config   config.Config
 }
 
+func (r *AmazonCloudWatchAgentReconciler) findCloudWatchAgentOwnedObjects(ctx context.Context, owner v1alpha1.AmazonCloudWatchAgent) (map[types.UID]client.Object, error) {
+	// Define a map to store the owned objects
+	ownedObjects := make(map[types.UID]client.Object)
+	selector := manifestutils.SelectorLabels(owner.ObjectMeta, "*")
+	delete(selector, "app.kubernetes.io/component") //ignore components
```

Comment:
nit: It's unlikely to change, but could we add a const for this key in `manifestutils` and use that instead?